entity-framework/core/miscellaneous/nativeaot-and-precompiled-queries.md

```diff
@@ -0,0 +1,110 @@
+author: roji
+ms.date: 11/10/2024
+uid: core/miscellaneous/nativeaot-and-precompiled-queries
+---
+# NativeAOT Support and Precompiled Queries (Experimental)
+
+> [!WARNING]
+> NativeAOT and query precompilation are highly experimental feature, and are not yet suited for production use. The support described below should be viewed as infrastructure towards the final feature, which will likely be released with EF 10. We encourage you to experiment with the current support and report on your experiences, but recommend against deploying EF NativeAOT applications in production. See below for specific known limitations.
+
+[.NET NativeAOT](/dotnet/core/deploying/native-aot) allows publishing self-contained .NET applications that have been compiled ahead-of-time (AOT). Doing so offers the following advantages:
+
+* Significantly faster application startup time
+* Small, self-contained binaries that have smaller memory footprints and are easier to deploy
+* Running applications in environments where just-in-time compilation isn't supported
+
+EF NativeAOT applications start up much faster than the same applications without NativeAOT; aside from the general startup improvements that NativeAOT offers (i.e. no JIT compilation required on each startup), EF's NativeAOT supports removes the processing of LINQ queries and their translation to SQL; the more EF LINQ queries an application has in its code, the faster the startup gains are expected to be from NativeAOT.
+
+## Publishing an EF NativeAOT Application
+
+First, enable NativeAOT publishing for your project as follows:
+
+```xml
+<PropertyGroup>
+    <PublishAot>true</PublishAot>
+</PropertyGroup>
+```
+
+EF's support for LINQ query execution under NativeAOT relies on *query precompilation*: this mechanism statically identifies EF LINQ queries and generates C# [*interceptors*](/dotnet/csharp/whats-new/csharp-12#interceptors), which contain code to execute each specific query. This can significantly cut down on your application's startup time, as the heavy lifting of processing and compiling your LINQ queries into SQL no longer happens every time your application starts up. Instead, each query's interceptor contains the finalized SQL for that query, as well as optimized code to materialize database results as .NET objects.
+
+C# interceptors are currently an experimental feature, and require a special opt-in in your project file:
+
+```xml
+<PropertyGroup>
+  <InterceptorsNamespaces>$(InterceptorsPreviewNamespaces);Microsoft.EntityFrameworkCore.GeneratedInterceptors</InterceptorsNamespaces>
+</PropertyGroup>
+```
+
+At this point, you're ready to precompile your LINQ queries, and generate the [compiled model](xref:core/performance/advanced-performance-topics#compiled-models) that they depend on. Make sure that you have at least version 9.0 of the EF tools (`dotnet tool list -g`), and then execute the following:
+
+```console
+dotnet ef dbcontext optimize --precompile-queries --nativeaot
+```
+
```

Comment:
I'll add a link to the MSBuild integration documentation here in my PR

Reply:
OK. I've amended the docs to show adding the Tasks package rather than running `dotnet ef dbcontext optimize` - in any case a link to the dedicated docs makes sense.